frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -570,19 +556,18 @@ export default ({
       if (msgIndex >= 0) {
         scrollAndHighlight(msgIndex)
       } else {
-        const contractID = this.summary.chatRoomID
+        const contractID = this.ephemeral.renderingChatRoomId
         const limit = this.chatRoomSettings?.actionsPerPage || CHATROOM_ACTIONS_PER_PAGE
         const events =
           // FIX: this.messages[0].height could not be the starting height of the events in the page
           await sbp('chelonia/out/eventsBetween', contractID, messageHash, this.messages[0].height, limit / 2, { stream: false })
             .catch((e) => {
               console.debug(`Error fetching events or message ${messageHash} doesn't belong to ${contractID}`, e)
             })
-        if (!this.checkEventSourceConsistency(contractID)) return
+
+        if (this.chatroomHasSwitchedFrom(contractID)) return
         if (events && events.length) {
           await this.rerenderEvents(events)
-
-          if (!this.checkEventSourceConsistency(contractID)) return
```

Comment:
We should probably add this back...

Reply:
Will do. be noted that the large number of these were already removed in [this code](https://github.com/okTurtles/group-income/issues/2556#issuecomment-2683819280) btw..